pkg/corerp/api/v20231001preview/environment_conversion_test.go

```diff
@@ -117,6 +117,19 @@ func TestConvertVersionedToDataModel(t *testing.T) {
 							Scope: "/planes/aws/aws/accounts/140313373712/regions/us-west-2",
 						},
 					},
+					RecipeConfig: datamodel.RecipeConfigProperties{
```

Comment:
This needs a few different layers and we could run into nil pointer exception easily. We should test it thoroughly.

Reply:
If we remove the PAT map from this test, we can see if we run into nil pointer exception above in the original conversion logic.